gammapy/datasets/simulate.py

```diff
@@ -70,25 +79,58 @@ def _evaluate_timevar_source(self, dataset, evaluator, time_axis=None):
 
         Returns
         -------
-        npred : `~gammapy.maps.Map`
+        npred : `~gammapy.maps.RegionNDMap`
             Npred map.
         """
-        energy_axis = MapAxis.from_edges(
-            dataset.geoms["geom"].axes["energy"].edges, name="energy_true"
+        energy_true = dataset.edisp.edisp_map.geom.axes["energy_true"]
+        energy_new = energy_true.upsample(10)
+        target = evaluator.model.spatial_model.position
+        edisp = dataset.edisp.get_edisp_kernel(
+            dataset.geoms["geom"].axes["energy"], position=target
         )
+        region_exposure = dataset.exposure.to_region_nd_map(target)
+
+        if not time_axis:
+            tstart = dataset.gti.time_start
+            tstop = dataset.gti.time_stop
+            nbin = int(((tstop - tstart) / t_delta).to(""))
+            time_axis_eval = TimeMapAxis.from_time_bounds(
+                time_min=tstart,
+                time_max=tstop,
+                nbin=nbin,
+            )
+            time_axis = MapAxis.from_bounds(
+                tstart[0].mjd * u.d,
+                tstop[0].mjd * u.d,
+                nbin=nbin,
+                name="time",
+            )
 
-        target = evaluator.model.spatial_model.position
-        on_region = PointSkyRegion(center=target)
-        region_geom = RegionGeom.create(on_region, axes=[energy_axis])
+        flux_diff = (
+            evaluator.model.temporal_model.evaluate(
+                time_axis_eval.time_mid, energy=energy_new.center
+            )
+            * evaluator.model.spectral_model.parameters[0].quantity
+        )
+
+        flux_inte = flux_diff * energy_new.bin_width[:, None]
+
+        f = interpolate.interp1d(energy_new.center, flux_inte, axis=0)
```

Comment:
Please avoid the "manual" interpolation here. You could create the `RegionNDMap` earlier and rely on `RegionNDMap.interpolate_by_coord()` instead.

Reply:
Done.